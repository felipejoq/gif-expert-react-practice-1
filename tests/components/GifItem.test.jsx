import {render, screen} from "@testing-library/react";
import {GifItem} from "../../src/components/index.js";

describe('Pruebas en <GifItem />', () => {

    const title = 'Arnold'
    const url = 'https://domain.test/image.jpg';

    test('debe hacer match con el snapshot.', () => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el url y alt indicado.', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el tÃ­tulo en el componente.', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
});