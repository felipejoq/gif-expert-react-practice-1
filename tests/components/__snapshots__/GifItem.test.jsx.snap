// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot. 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Arnold"
      src="https://domain.test/image.jpg"
    />
    <p>
      Arnold
    </p>
  </div>
</div>
`;
